lib/src/rules/unnecessary_brace_in_string_interps.dart

```diff
@@ -68,6 +68,8 @@ class _Visitor extends SimpleAstVisitor<void> {
             !identifier.name.contains('\$')) {
           rule.reportLint(expression);
         }
+      } else if (exp is ThisExpression) {
+        rule.reportLint(expression);
```

Comment:
Some of the same tests as for `SimpleIdentifier` need to be done here. Consider, for example `'ab${this}cd'`. Removing the braces there would be wrong. You might even be able to use a single branch for both cases.

Reply:
Ah!  Good catch.  Thank you!

Reply:
Done. 👍